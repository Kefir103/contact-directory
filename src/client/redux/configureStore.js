import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
