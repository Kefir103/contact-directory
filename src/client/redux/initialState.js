export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
