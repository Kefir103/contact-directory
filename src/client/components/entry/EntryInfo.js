import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { changeElementDescription } from '../../redux/actions/dataActions';

const EntryInfo = (props) => {
    const handleDescriptionChange = (event) => {
        props.actions.changeElementDescription(event.target.value, props.elementIndex);
    };

    return (
        <div className={'entry-info'}>
            <p>
                Выбран пользователь:{' '}
                <b>
                    {props.element.firstName} {props.element.lastName}
                </b>
            </p>
            <p>Описание:</p>
            <textarea
                id={'textarea-info-description'}
                onChange={handleDescriptionChange}
                value={props.element.description}
            />
            <p>
                Адрес проживания: <b>{props.element.address.streetAddress}</b>
            </p>
            <p>
                Город: <b>{props.element.address.city}</b>
            </p>
            <p>
                Провинция/штат: <b>{props.element.address.state}</b>
            </p>
            <p>
                Индекс: <b>{props.element.address.zip}</b>
            </p>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        element: state.data.currentElement.elementInfo,
        elementIndex: state.data.currentElement.elementIndex,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(
            {
                changeElementDescription,
            },
            dispatch
        ),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(EntryInfo);
