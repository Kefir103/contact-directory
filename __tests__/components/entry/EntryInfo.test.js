import React from 'react';
import configureMockStore from 'redux-mock-store';
import EntryInfo from '../../../src/client/components/entry/EntryInfo';
import { shallow, render } from 'enzyme';
import { initialState } from '../../../src/client/redux/initialState';

const mockStore = configureMockStore();

describe('<EntryInfo />', () => {
    test('should render correctly with full ElementInfo', () => {
        const currentElement = {
            elementInfo: {
                id: 1,
                firstName: 'firstName',
                lastName: 'lastName',
                emaiL: 'email',
                phone: 'phone',
                address: {
                    streetAddress: 'streetAddress',
                    city: 'city',
                    state: 'state',
                    zip: 'zip',
                },
                description: 'description',
            },
            elementIndex: 1,
        };
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                currentElement,
            },
        });
        const wrapper = shallow(<EntryInfo store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should render correctly without address in ElementInfo', () => {
        const currentElement = {
            elementInfo: {
                id: 1,
                firstName: 'firstName',
                lastName: 'lastName',
                emaiL: 'email',
                phone: 'phone',
                description: 'description',
            },
            elementIndex: 1,
        };
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                currentElement,
            },
        });
        const wrapper = shallow(<EntryInfo store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
});
