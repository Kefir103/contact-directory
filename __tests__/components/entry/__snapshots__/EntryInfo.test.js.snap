// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryInfo /> should render correctly with full ElementInfo 1`] = `
<div
  class="entry-info"
>
  <p>
    Выбран пользователь: 
    <b>
      firstName lastName
    </b>
  </p>
  <p>
    Описание:
  </p>
  <textarea
    id="textarea-info-description"
  >
    description
  </textarea>
  <p>
    Адрес проживания: 
    <b>
      streetAddress
    </b>
  </p>
  <p>
    Город: 
    <b>
      city
    </b>
  </p>
  <p>
    Провинция/штат: 
    <b>
      state
    </b>
  </p>
  <p>
    Индекс: 
    <b>
      zip
    </b>
  </p>
</div>
`;

exports[`<EntryInfo /> should render correctly without address in ElementInfo 1`] = `
<div
  class="entry-info"
>
  <p>
    Выбран пользователь: 
    <b>
      firstName lastName
    </b>
  </p>
  <p>
    Описание:
  </p>
  <textarea
    id="textarea-info-description"
  >
    description
  </textarea>
</div>
`;
